src/util/cilfacade.ml

```diff
@@ -85,6 +85,18 @@ let do_preprocess ast =
   in
   iterGlobals ast (function GFun (fd,_) -> List.iter (f fd) !visitors | _ -> ())
 
+let visitors_cil = ref []
+(* does exactly the same as register_preprocess but it is executed earlier, before the CFG is created*)
+let register_preprocess_cil name visitor_fun =
+  visitors_cil := !visitors_cil @ [name, visitor_fun]
+
+let do_preprocess_cil ast =
+  let f fd (name, visitor_fun) =
+    (* this has to be done here, since the settings aren't available when register_preprocess is called *)
+    if List.mem name (get_string_list "ana.activated") then
+      ignore @@ visitCilFunction (visitor_fun fd) fd
+  in
+  iterGlobals ast (function GFun (fd,_) -> List.iter (f fd) !visitors_cil | _ -> ())
```

Comment:
Should just fix `register_preprocess` then. It's only used by the old termination analysis, which probably also needs this to happen before CFG, no? It might be that `do_preprocess` was at one point accidentally moved around and nothing we test uses it.